src/Pos/Block/Logic.hs

```diff
@@ -751,16 +750,18 @@ createMainBlockPure
     -> SscPayload ssc
     -> UpdatePayload
     -> SecretKey
-    -> MainBlock ssc
+    -> m (MainBlock ssc)
 createMainBlockPure limit prevHeader txs pSk sId psks sscData usPayload sk =
-    flip evalState limit $ do
+    flip evalStateT limit $ do
```

Comment:
[`evaluatingStateT`](https://github.com/serokell/universum/blob/9edb5d374ec1fb6bb873dacc22a013f1eacd9c67/src/Monad/Trans.hs#L59) can be used here